src/modules/commander/Commander.hpp

```diff
@@ -122,6 +114,32 @@ class Commander : public ModuleBase<Commander>, public ModuleParams
 
 	void mission_init();
 
+	/**
+	 * Update the telemetry status and the corresponding status variables.
+	 * Perform system checks when new telemetry link connected.
+	 */
+	void poll_telemetry_status(bool checkAirspeed, bool *hotplug_timeout);
+
+	/**
+	 * Checks the status of all available data links and handles switching between different system telemetry states.
+	 */
+	void data_link_checks(int32_t highlatencydatalink_loss_timeout, int32_t highlatencydatalink_regain_timeout,
+			int32_t datalink_loss_timeout, int32_t datalink_regain_timeout, bool *status_changed);
+
+	// telemetry variables
+	int _telemetry_subs[ORB_MULTI_MAX_INSTANCES] = {};
```

Comment:
can you create a separate struct for these? Then you need only one array and can use a constructor or member initializer to initialize the fields

Reply:
Good point. I added that struct.